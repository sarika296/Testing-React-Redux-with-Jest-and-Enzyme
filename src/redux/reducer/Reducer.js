import Action from '../action/actionTypes'
 
export const initialState = {
    numOfCakes: 10,
    numOfIceCreams: 20,
    numOfChocolates: 30,
    totalMoney: 0
}

const Reducer = (state = initialState, action ) => {
    switch(action.type) {
        case Action.BUY_CAKE : return {
            ...state,
            numOfCakes: state.numOfCakes - action.payload,
            totalMoney: state.totalMoney + action.cost
        }

        case Action.BUY_CHOCOLATE : return {
            ...state,
            numOfChocolates: state.numOfChocolates - action.payload,
            totalMoney: state.totalMoney + action.cost
        }

        case Action.BUY_ICECREAM: return {
            ...state,
            numOfIceCreams: state.numOfIceCreams - action.payload,
            totalMoney: state.totalMoney + action.cost
        }

        default: return state
    }
}

export default Reducer 