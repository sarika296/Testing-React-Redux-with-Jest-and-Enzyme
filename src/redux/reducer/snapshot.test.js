import React from 'react'
import renderer from 'react-test-renderer'
import Reducer from '../../App'

describe('Reducer Snapshot Testing...', () => {
    it('renders component correctly', () => {
        const tree = renderer.create(<Reducer />).toJSON();
        expect(tree).toMatchSnapshot();
    });
}); 
