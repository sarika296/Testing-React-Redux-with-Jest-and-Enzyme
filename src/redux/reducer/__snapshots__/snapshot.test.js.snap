// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reducer Snapshot Testing... renders component correctly 1`] = `
Array [
  <div
    className="cake"
  >
    Cake Shop App
  </div>,
  <div
    className="App"
  >
    <div>
      <h2>
        Number of cakes - 
        10
         
      </h2>
      <input
        onChange={[Function]}
        type="text"
        value={1}
      />
      <button
        onClick={[Function]}
      >
        Buy 
        1
         Cake
      </button>
    </div>
    <div>
      <h2>
        Number of ice-creams - 
        20
         
      </h2>
      <input
        onChange={[Function]}
        type="text"
        value={1}
      />
      <button
        onClick={[Function]}
      >
        Buy 
        1
         Ice Cream
      </button>
    </div>
    <div>
      <h2>
        Number of chocolates - 
        30
         
      </h2>
      <input
        onChange={[Function]}
        type="text"
        value={1}
      />
      <button
        onClick={[Function]}
      >
        Buy 
        1
         Chocolate
      </button>
    </div>
    <div>
      <h2>
        Total Money To The Counter
      </h2>
      <p
        style={
          Object {
            "fontSize": 20,
          }
        }
      >
        Total Money - 
        0
         
      </p>
    </div>
  </div>,
]
`;
